subsystem/kubeapi_test.go

```diff
@@ -3279,84 +3279,216 @@ spec:
 	})
 })
 
-var _ = Describe("bmac reconcile flow", func() {
+var _ = Describe("bmac reconcile ztp flow", func() {
 	if !Options.EnableKubeAPI {
 		return
 	}
 
 	ctx := context.Background()
 
 	var (
-		bmh         *bmhv1alpha1.BareMetalHost
-		bmhNsName   types.NamespacedName
-		agentNsName types.NamespacedName
-		infraNsName types.NamespacedName
+		clusterDeploymentSpec *hivev1.ClusterDeploymentSpec
+		infraEnvSpec          *v1beta1.InfraEnvSpec
+		infraNsName           types.NamespacedName
+		aciSpec               *hiveext.AgentClusterInstallSpec
+		secretRef             *corev1.LocalObjectReference
+		bmh                   *bmhv1alpha1.BareMetalHost
+		//bmhNsName             types.NamespacedName
+		//agentNsName           types.NamespacedName
 	)
 
 	BeforeEach(func() {
-		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
-		clusterDeploymentSpec := getDefaultClusterDeploymentSpec(secretRef)
-		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
-
+		secretRef = deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		clusterDeploymentSpec = getDefaultClusterDeploymentSpec(secretRef)
+		aciSpec = getDefaultAgentClusterInstallSpec(clusterDeploymentSpec.ClusterName)
 		infraNsName = types.NamespacedName{
 			Name:      "infraenv",
 			Namespace: Options.Namespace,
 		}
-		infraEnvSpec := getDefaultInfraEnvSpec(secretRef, clusterDeploymentSpec)
-		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
-
-		infraEnvKey := types.NamespacedName{
-			Namespace: Options.Namespace,
-			Name:      infraNsName.Name,
-		}
-		infraEnv := getInfraEnvFromDBByKubeKey(ctx, db, infraEnvKey, waitForReconcileTimeout)
-		configureLocalAgentClient(infraEnv.ID.String())
-		host := registerNode(ctx, *infraEnv.ID, "hostname1", defaultCIDRv4)
-		agentNsName = types.NamespacedName{
-			Namespace: Options.Namespace,
-			Name:      host.ID.String(),
-		}
-
-		bmhSpec := bmhv1alpha1.BareMetalHostSpec{}
-		deployBMHCRD(ctx, kubeClient, host.ID.String(), &bmhSpec)
-		bmhNsName = types.NamespacedName{
-			Namespace: Options.Namespace,
-			Name:      host.ID.String(),
-		}
+		infraEnvSpec = getDefaultInfraEnvSpec(secretRef, clusterDeploymentSpec)
+		deployClusterImageSetCRD(ctx, kubeClient, aciSpec.ImageSetRef)
 	})
 
 	AfterEach(func() {
 		cleanUpCRs(ctx, kubeClient)
+		verifyCleanUP(ctx, kubeClient)
 		clearDB()
 	})
 
-	Context("sno reconcile flow", func() {
-		It("reconciles the infraenv", func() {
-			bmh = getBmhCRD(ctx, kubeClient, bmhNsName)
-			bmh.SetLabels(map[string]string{controllers.BMH_INFRA_ENV_LABEL: infraNsName.Name})
-			Expect(kubeClient.Update(ctx, bmh)).ToNot(HaveOccurred())
+	Context("bmac reconcile flow", func() {
+		It("create spoke cluster and make it day2 adding new bmh using ztp flow", func() {
+			By("Create cluster")
+			deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+			deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+			time.Sleep(time.Second * 10)
```

Comment:
Do not use Sleep, it will result in flaky test.
If you need to wait to something that should be created/updated, use `Eventually`